### PR TITLE
Feature/4.0

### PR DESCRIPTION
**Pull request by konstantin24121**
### New features
- `includeKeys`, for performance. Know more about this feature in README
- new hook `useIsolatedStore`
- new overloads for `useStore` hook. Short way to watch your stores.
### Added
- optional `uniqKey` added into StoreOptions
- clearPersistence function for cleaning state into storage
- new demo pages
- tests for new features

### CODE REVIEW

**Comment by konstantin24121** [comment]
@ruslanchek I guess it is ready for release and publish

File: lib/index.d.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
...
-export { StoreEvent, StoreEventType } from './StoreEvent';
-export { StorePersistentDriver } from './StorePersistentDriver';
+export * from './followStore';
+export * from './Store';
+export * from './StorePersistentLocalStorageDriver';
+export * from './useStore';
+export * from './useIsolatedStore';
+export { StoreEvent, StoreEventType } from './StoreEvent';
+export { StorePersistentDriver } from './StorePersistentDriver';
+export * from './compare';
```
Comment (ruslanchek) [inline lib/index.d.ts]: How do you plan to use it outside of the module?

File: src/useStore.ts [hunk 2]
```diff
@@ -1,34 +1,134 @@
...
-  const [state, setState] = React.useState(mapState(store.state));
+function getOption<T, V>(rest: Array<any>): Required<IOptionsExtended<T, V>> {
+  const mapState = (store: T) => store as V & T;
+  const compare = (a: V, b: V) => a === b;
+  const eventType = StoreEventType.All;
+  const includeKeys = [];
+  if (
+    rest[0] &&
+    (Object.keys(StoreEventType).indexOf(rest[0].toString()) >= 0 ||
+      (Array.isArray(rest[0]) && Object.keys(StoreEventType).indexOf(rest[0][0].toString()) >= 0))
```
Comment (pret-a-porter) [inline src/useStore.ts]: Думаю, тут `indexOf` можно заменить на `includes`

File: README.md [hunk 3]
```diff
@@ -187,57 +193,104 @@ import { myStore } from './myStore';
 console.log(myStore.state.counter); // 9999
 ```
 
-## API
+### useIsolatedStore
+
+```typescript tsx
+import React, { FC } from 'react';
+import { useStore } from 'react-stores';
```
Comment (pret-a-porter) [inline README.md]: `useStore` => `useIsolatedStore `

Comment (konstantin24121) [reply README.md]: Thanks, fixed

File: lib/index.d.ts (hunk 1 above)
Comment (konstantin24121) [reply lib/index.d.ts]: We exported `areSimilar` function from `./compare`  for lazy developers to equality prev and next states. In [README](https://github.com/ibitcy/react-stores/pull/31/files#diff-04c6e90faac2675aa89e2176d2eec7d8R569) we mention this.

`./useIsolatedStore` exports a new hook.

`./StorePersistentDriver` exports interfaces and abstractions for creating drivers for persistent stores

`StoreEventType` also need for  using `on(StoreEventType.Something, ...`

Maybe only `StoreEvent` exporting is not necessary  🤷🏽

Comment (ruslanchek) [reply lib/index.d.ts]: `StoreEvent` is really unnecessary, but one of our teammates asked me to export it. So we hold it for a while.

File: src/useStore.ts (hunk 2 above)
Comment (ruslanchek) [reply src/useStore.ts]: Sure! It's even also way faster after all.

Comment (konstantin24121) [reply src/useStore.ts]: Fixed

Comment (konstantin24121) [reply src/useStore.ts]: Ops, fixed in another file

Comment (konstantin24121) [reply src/useStore.ts]: Fixed

**Pull request merged**